src/main/java/de/tum/in/www1/artemis/web/rest/FileUploadSubmissionResource.java

```diff
@@ -257,11 +255,7 @@ public FileUploadSubmissionResource(SubmissionRepository submissionRepository, R
         else {
             Optional<FileUploadSubmission> optionalFileUploadSubmission = fileUploadSubmissionService
                     .getRandomFileUploadSubmissionEligibleForNewAssessment((FileUploadExercise) fileUploadExercise, fileUploadExercise.isExamExercise(), correctionRound);
-
-            if (optionalFileUploadSubmission.isEmpty()) {
-                return notFound();
-            }
-            fileUploadSubmission = optionalFileUploadSubmission.get();
+            fileUploadSubmission = optionalFileUploadSubmission.orElseThrow(() -> new EntityNotFoundException("fileUploadSubmission without Assessment"));
```

Comment:
```suggestion
            fileUploadSubmission = optionalFileUploadSubmission.orElseThrow(() -> new EntityNotFoundException("File Upload Submission without Assessment"));
```
Would match the style in other places (e.g. your new `FileUploadExerciseRepository` method) better.